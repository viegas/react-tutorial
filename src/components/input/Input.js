import React from 'react';
import PropTypes from 'prop-types';
import styles from './Input.module.css';

const Input = ({ onSelectAll, onCreateTodo }) => {
    const handleKeyPress = ({ key, target }) => {
        if (key === 'Enter' && target.value !== '') {
            onCreateTodo(target.value);
            target.value = '';
        }
    };

    return (
        <div className={styles.wrapper}>
            <input type="checkbox" onClick={onSelectAll} />
            <input
                type="text"
                className={styles.input}
                placeholder={'What needs to be done?'}
                onKeyPress={handleKeyPress}
            />
        </div>
    );
};

Input.propTypes = {
    onSelectAll: PropTypes.func,
    onCreateTodo: PropTypes.func,
};

Input.defaultProps = {
    onSelectAll: () => {
        console.log('onSelectAll');
    },
    onCreateTodo: (value) => {
        console.log('onCreateTodo', value);
    },
};

export default Input;
