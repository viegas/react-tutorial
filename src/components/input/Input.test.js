import Input from './Input';

it('renders correctly', () => {
    const wrapper = shallow(
        <Input onSelectAll={() => {}} onCreateTodo={() => {}} />
    );

    expect(wrapper).toMatchSnapshot();
});

// it('Component should handle onChange function', () => {
//     const wrapper = shallow(<Input />);

//     const evt = {
//         target: {
//             value: 'Some Value',
//         },
//     };

//     wrapper.find('input').simulate('change', evt);

//     expect(wrapper.find('input').prop('value')).toEqual('Some Value')
//     expect(wrapper).toMatchSnapshot();

// });
