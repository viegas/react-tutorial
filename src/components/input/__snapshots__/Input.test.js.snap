// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="wrapper"
>
  <input
    onClick={[Function]}
    type="checkbox"
  />
  <input
    className="input"
    onKeyPress={[Function]}
    placeholder="What needs to be done?"
    type="text"
  />
</div>
`;
